modules/generic_worker/manifests/init.pp

```diff
@@ -118,22 +121,14 @@
                         owner   => $users::root::username,
                         group   => $users::root::group;
                     }
-                    exec { 'create gpg key':
-                        path    => ['/bin', '/sbin', '/usr/local/bin', '/usr/bin'],
-                        user    => 'cltbld',
-                        cwd     => $users::builder::home,
-                        command => "generic-worker new-openpgp-keypair --file ${::users::builder::home}/generic-worker.openpgp.key",
-                        unless  => "test -f ${::users::builder::home}/generic-worker.openpgp.key"
-                    }
-
                     host {"${taskcluster_host}":
                         ip => '127.0.0.1'
                     }
                     # Enable proxy_http_module on apache2
                     exec { 'enable proxy_http_module':
-                        path    => ['/bin', '/sbin', '/usr/local/bin', '/usr/bin', '/usr/sbin'],
-                        command => 'a2enmod proxy_http',
-                        notify  => Service['httpd'];
+                        path      => ['/bin', '/sbin', '/usr/local/bin', '/usr/bin', '/usr/sbin'],
+                        command   => 'a2enmod proxy_http',
+                        subscribe => Service['httpd'];
```

Comment:
I think 'notify' was the correct relationship metaparameter to set here.  Since a2enmod sets up symbolic links in the config directories, I believe you will want the service to be restarted _**after**_ the a2enmod has executed.  My previous comments might have been a little confusing.  Sorry about that.

You should also have an 'onlyif, unless, or creates' to prevent this from executing on every run.  I would suggest using 'creates' to check for the symlink in /etc/apache2/mods-enabled

http://manpages.ubuntu.com/manpages/bionic/man8/a2enmod.8.html